sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/Serializer.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.impl
+
+import com.akkaserverless.javasdk.JsonSupport
+// FIXME: should we duplicate this in the scala sdk and expose ScalaPbAny instead?
+import com.google.protobuf.{ Any => JavaPbAny }
+
+object Serializer {
+
+  def noopSerializer: Serializer = NoopSerializer
+
+  object NoopSerializer extends Serializer {
+    override def canSerialize(any: Any): Boolean = false
+
+    override def serialize(any: Any): JavaPbAny =
+      throw new IllegalArgumentException("NoopSerializer can not serialize types")
+
+    override def canDeserialize(any: JavaPbAny): Boolean = false
+
+    override def deserialize(any: JavaPbAny): Any =
+      throw new IllegalArgumentException("NoopSerializer can not deserialize types")
+
+  }
+
+}
+trait Serializer {
```

Comment:
I originally called it a Codec because the JsonSupport and AnySupport has methods like `encode` / `decode`, but after looking at it once more I thought that it's more clear to users if we just call it a `Serializer`. 

Users are not expected to implement `Serializers` though, but they can if they want. Actually, they could use any format now, not only Json. 

Which makes me think that we should default to cbor instead. 

We could expand the annotation and let the user chose some pre-defined serialization formats, `cbor`, `json` or `custom`.  Where `custom` means, we don't generate anything and the user will need to write their serializers and configure the provider accordingly.

Reply:
I don't think we should open up for generic serialization, or support multiple formats, instead we should make sure that the format we _do_ support (`json`) works seamless throughout Akka Serverless, for example using the `json.akkaserverless.com` type url for the Any's and then special-handling that where needed in the proxy and SDKs.

Reply:
Agree that a custom serialization may not be needed and even complicated things. That was not my intention to add it right now anyway. 

I do think that cbor can be interesting eventually since we do that in akka as well, but again, nothing that we should care about now.